jina/serve/networking.py

```diff
@@ -315,15 +318,21 @@ def __init__(
             self,
             address,
             channel,
+            deployment_name: str,
             metrics: _NetworkingMetrics,
             histograms: _NetworkingHistograms,
         ):
             self.address = address
             self.channel = channel
+            self.deployment_name = deployment_name
             self._metrics = metrics
             self._histograms = histograms
             self._initialized = False
 
+            if self._histograms is not None:
+                if self._histograms.histogram_metric_labels is not None:
+                    self._histograms.histogram_metric_labels.update(deployment_name=deployment_name)
```

Comment:
Doing update on the histogram_metric_labels might be wrong because it is passed by reference. Make sure to check this